### ISSUE TITLE
Support specifying required language version

### ISSUE THREAD

**Issue by diwakergupta** [clarity-lang/reference#12]
A common feature in many languages allows programs to specify the minimum language / runtime version required to run that program. A very useful safeguard, particularly in a blockchain environment.

**Comment by obycode** [comment]
Yes, this would be helpful to see in the code, especially if new Clarity versions change the syntax or semantics of existing language constructs. Without it, you need to lookup the deployment of the contract to see what version of Clarity was specified. I think it could be done with a magic comment, maybe something like:

```clarity
;; #[clarity(4)]
```

If the version specified here does not match the version specified in the deploy transaction, the deploy should fail.
